### PR TITLE
Linux kernel fuzzing example

### PR DESCRIPTION
**Pull request by rmalmain**
add an example for linux kernel systemmode fuzzing.

### CODE REVIEW

**Comment by rmalmain** [comment]
can you check if the `fuzzers` structure looks good @domenukk ?

**Comment by rmalmain** [comment]
i'd say it's like inprocess vs binary-only, fuzzbench is like a more "specific" version of inprocess

**Comment by domenukk** [comment]
We could just put them into inprocess and maybe add some extra Readme in there?

**Comment by rmalmain** [comment]
> We could just put them into inprocess and maybe add some extra Readme in there?

thoughts @tokatoka ? if i remember well you preferred to separate fuzzbench fuzzers, no?

**Comment by tokatoka** [comment]
not all fuzzbench fuzzers are in-process fuzzers

**Comment by rmalmain** [comment]
there are also some forkserver fuzzers in the fuzzbench directory

**Comment by domenukk** [comment]
Let's move the forkserver fuzzbench fuzzers into forkserver then?

**Comment by domenukk** [comment]
(Most people don't know what fuzzbench means I assume, it's just another fuzzer I would argue)

**Comment by rmalmain** [comment]
if everyone is fine with this i'll move fuzzbench fuzzers to corresponding directories

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -273,29 +273,27 @@ jobs:
           - ./fuzzers/binary-only/frida_executable_libpng
           - ./fuzzers/binary-only/frida_gdiplus
           - ./fuzzers/binary-only/frida_libpng
+          - ./fuzzers/binary-only/fuzzbench_qemu
           - ./fuzzers/binary-only/tinyinst_simple
 
           # Forkserver
           - ./fuzzers/forkserver/forkserver_simple
           - ./fuzzers/forkserver/forkserver_libafl_cc
+          - ./fuzzers/forkserver/fuzzbench_fork_qemu
```
Comment (domenukk) [inline .github/workflows/build_and_test.yml:282]: That guy is probably also binary-only (?)

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -273,29 +273,27 @@ jobs:
...
+          - ./fuzzers/forkserver/fuzzbench_forkserver
+          - ./fuzzers/forkserver/fuzzbench_forkserver_cmplog
           - ./fuzzers/forkserver/libafl-fuzz
 
           # Full-system
```
Comment (domenukk) [inline .github/workflows/build_and_test.yml:287]: What about the qemu full system example?

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (rmalmain) [reply .github/workflows/build_and_test.yml:282]: yes it is, i don't know which should be used

Comment (domenukk) [reply .github/workflows/build_and_test.yml:282]: I would say it's more important that it's binary only then
Like, full system > binary only > inprocess/forkserver

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (rmalmain) [reply .github/workflows/build_and_test.yml:287]: they should all be in Full-system. someone is missing?

Comment (rmalmain) [reply .github/workflows/build_and_test.yml:287]: ah they are a bit below, to run with the qemu-compatible runners

**Comment by tokatoka** [comment]
> if everyone is fine with this i'll move fuzzbench fuzzers to corresponding directories


it's ok

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (rmalmain) [reply .github/workflows/build_and_test.yml:282]: from more to less specific so, i'm fine with this

**Comment by rmalmain** [comment]
what i didn't do in this pr is the modification of the EDGES_MAP_* stuff.
i'll do the modifications in another pr.

**Comment by rmalmain** [comment]
this pr is ready, i'll merge when the ci is fixed

**Comment by domenukk** [comment]
IMHO libafl_atheris should be in an `other_target` folder (or similar). It's specifically for fuzzing python.

**Comment by rmalmain** [comment]
> IMHO libafl_atheris should be in an `other_targets` folder (or similar). It's specifically for fuzzing python.
what else could we put there

> And it might make sense to have one folder for better/different input generations like nautilus (`structured_inputs`?)
i'd use more smth like grammar-aware, no? usually people know more about grammar fuzzing right?

**Pull request merged**